runtime/src/snapshot_info.rs

```diff
@@ -0,0 +1,173 @@
+//! Shared runtime information about snapshots
+//!
+//! Snapshot information is needed at runtime in multiple threads.  Currently, this is used to
+//! prevent cleaning of zero-lamport accounts in `AccountsDb` _past_ the last full snapshot slot.
+//! However, snapshots are taken in a separate thread, and previously `AccountsDb` didn't have a
+//! way to get that information.
+//!
+//! This module was created to solve that problem.  Now, all threads that need runtime snapshot
+//! information can have a `SyncSnapshotInfo` object to get/set the information they need.
+//!
+//! **NOTE** Client should use `SyncSnapshotInfo` and not `SnapshotInfo`, since the later does not
+//! include synchronization (and the whole point of this module is to share snapshot information
+//! between threads).
+//!
+//! **NOTE** There are helper functions to handle the common case of getting/setting snapshot
+//! information when you have a `Option<SyncSnapshotInfo>`.
+
+use solana_sdk::clock::Slot;
+use std::sync::{Arc, RwLock};
+
+/// Runtime information about snapshots
+///
+/// **NOTE** Clients should use `SyncSnapshotInfo` instead!
+///
+/// Information about the snapshots is needed in multiple threads at different times; for example,
+/// `AccountsDb` (and `Bank` via `AccountsDb`) need to ensure cleaning does not happen past the
+/// `last_full_snapshot_slot`.  Whereas `snapshot_utils`, `serde_snapshot`, `AccountsBackgroundService`, and
+/// `SnapshotPackagerService` need to be able to set these values when snapshots are taken.
+///
+/// Share this object between all interested parties to ensure a consistent view of the snapshot
+/// information.
+#[derive(Debug)]
+pub struct SnapshotInfo {
+    /// The slot when the last full snapshot was taken.  Can be `None` if there are no full
+    /// snapshots yet.
+    last_full_snapshot_slot: Option<Slot>,
+
+    /// The slot when the last incremental snapshot was taken.  Can be `None` if there are no
+    /// incremental snapshots yet.  It is required that this incremental snapshot's base slot is
+    /// the same as the `last_full_snapshot_slot`.
+    last_incremental_snapshot_slot: Option<Slot>,
```

Comment:
might be easier to just make these `pub` fields instead of providing getters and setters.  Some nuances explained here: https://users.rust-lang.org/t/public-getter-method-vs-pub-field/20147

Reply:
Yah, I agree. I wanted to make the "set" case a little better. Since the field is wrapped in an `Option` right now, if you're setting a "last slot" value, then it's never going to be `None` again, so clearing the Option/setting back to `None` should be illegal. Does that seem worth it in this case?

Reply:
Yeah if there's integrity checks, I think the getter/setter methods are worth it!